test/source/singleton_test.cpp

```diff
@@ -0,0 +1,46 @@
+#include <doctest/doctest.h>
+
+#include <iostream>
+#include <vector>
+
+#include <xsparse/levels/compressed.hpp>
+#include <xsparse/levels/singleton.hpp>
+
+TEST_CASE("Singleton-COO")
+{
+    constexpr uintptr_t SIZE = 100;
+    constexpr uintptr_t SIZE1 = 100;
```

Comment:
Can we test on corner cases like SIZE = 100, SIZE1 = 0 or SIZE = 1, SIZE1 = 100? Not sure about the semantics of this test but my intent is to make sure that added code works fine with whatever corner cases are possible. Thanks.

Reply:
> Can we test on corner cases like SIZE = 100, SIZE1 = 0 or SIZE = 1, SIZE1 = 100? Not sure about the semantics of this test but my intent is to make sure that added code works fine with whatever corner cases are possible. Thanks.

Yes that can be done, but as of now, SIZE by itself does not participate in the functions of singleton (pos_bounds and pos_access) so the value of SIZE should not affect the outcome of the test case. 
Maybe, I can add another test case having SIZE as 0 and SIZE1 as 100?

Reply:
Yes I would prefer to do so.

Reply:
I added a test case for a 3rd order COO format which has a compressed level followed by two singleton levels. I've used 0, 100 and 1 as edge case sizes.

Reply:
Please feel free to resolve this thread.